app/models/user.rb

```diff
@@ -470,6 +473,15 @@ def find_ids(options)
 
   protected
 
+  def promote_to_first_admin
+    account.update({first_admin_id: id})
+  end
+
+  def update_account_first_admin
+    return unless (other_admin = account.admins.where.not(id: id).first)
```

Comment:
Will the user be deleted if there’s no `other_admin` to take its place – i.e. to be promoted to first? How about nullify the account’s `first_admin_attribute` attribute in this case? I’m concerned about leaving behind this phantom id there pointing to a user that no longer exists.

Reply:
We need it for the events. It is used as a Key in the 3rd party service we use.

Reply:
I think we need to leave track of this conversation.

I told @guicassolato in Slack:

> This is exactly the goal of all this, to don't ever have null there 😄 
> Please read https://github.com/3scale/porta/issues/10#issuecomment-441031162 and https://github.com/3scale/porta/issues/10#issuecomment-441288361

He said:

> Yeah, but is that worse than having an ID to non-existent user?

I said:

> Yes. Because then the Events are not created and the actions are not executed. And having an ID with a non-existent user is temporary... it takes minutes (or less) to remove the account after that.

@hallelujah said:

> We need to have this ID. We should not add too much logic on this one: meaning relying on it for other thing than for events.


I think @guicassolato and I will have a call in Slack about this 👍

Reply:
We spoke in Slack. @guicassolato was lacking this information: We can not delete the last admin user alone without deleting the account 😄 
So all clear now 👍